asyncio_run_in_process/run_in_process.py

```diff
@@ -185,10 +184,17 @@ def open_in_process(
     loop: asyncio.AbstractEventLoop = None,
     subprocess_kwargs: 'SubprocessKwargs' = None,
 ) -> AsyncContextManager[ProcessAPI[TReturn]]:
-    return cast(
-        AsyncContextManager[ProcessAPI[TReturn]],
```

Comment:
Looks like this and the update to `mypy` could get extracted into something standalone to slightly reduce the overall size of this PR.

Reply:
Ideally I'd have done that, but one of the new extensions causes the new mypy to be installed and CI to fail, so I just went ahead and updated our version as well